sql/core/src/main/scala/org/apache/spark/sql/execution/ColumnarBatchScan.scala

```diff
@@ -93,7 +93,7 @@ private[sql] trait ColumnarBatchScan extends CodegenSupport {
     }
 
     val nextBatch = ctx.freshName("nextBatch")

```

Comment:
actually, how about we make `addNewFunction` accepts `funcNameHint` and generate unique func name inside `addNewFunction`? Then users can just call: `val nextBatch = ctx.addNewFunction("nextBatch", ...)`